### ISSUE TITLE
Error: Debug Failure typeToString

### ISSUE THREAD

**Issue by gp187** [Microsoft/TypeScript#17369]
**TypeScript Version:**  2.4.1

**Code**

```ts
// Doesn't work! 
export class Parent {
    public getThisStuff(): {s: [], cs: []} {
        let exportNames = {s: [], cs: []};

        return exportNames;
    }
}
// Doesn't work!
export class Parent {
    public getThisStuff() {
        let exportNames: {s: [], cs: []} = {s: [], cs: []};

        return exportNames;
    }
}
// works
export class Parent {
    public getThisStuff() {
        let exportNames = {s: [], cs: []};

        return exportNames;
    }
}
```

**Expected behavior:**
_a simple class, should work just fine_

**Actual behavior:**
```
/lab/o0/cli2/node_modules/typescript/lib/tsc.js:55776
                throw e;
                ^

Error: Debug Failure. 
    at typeToString (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:23526:22)
    at reportRelationError (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:28855:34)
    at isRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:28972:21)
    at propertiesRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:29382:43)
    at structuredTypeRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:29275:38)
    at recursiveTypeRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:29172:53)
    at isRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:28959:38)
    at checkTypeRelatedTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:28838:26)
    at checkTypeAssignableTo (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:28578:20)
    at checkReturnStatement (/lab/o0/cli2/node_modules/typescript/lib/tsc.js:37580:29)
```

**Comment by ikatyang** [comment]
You have to give array type a base type, e.g. `any[]`, `never[]`, etc.

For the Debug Failure, should be fixed by #17311

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
